plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/SyncPartitionMetadataProcedure.java

```diff
@@ -191,7 +191,7 @@ private static Set<Location> listDirectories(TrinoFileSystem fileSystem, Locatio
     private static String listedDirectoryName(Location directory, Location location)
     {
         String prefix = directory.path();
-        if (!prefix.endsWith("/")) {
+        if (!prefix.endsWith("/") && !prefix.equals("")) {
```

Comment:
Can we have some test coverage for this change?

Reply:
Thank you for comment..!

I cannot find any existing test suites for built-in procedures in Hive plugin to add some test case for this hotfix.
- https://github.com/trinodb/trino/tree/master/plugin/trino-hive/src/test/java/io/trino/plugin/hive

Did you mean creating new test module for this procedure and add all the test cases for it..?
(Is it okay to maintain other hive procedures without test suites..?)

Reply:
We have so called product tests that can be used to check the accuracy of `sync_partition_metadata` procedure

https://github.com/trinodb/trino/blob/33dd20a8c104d358f5b6d38e8a0405f8b0ced944/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestSyncPartitionMetadata.java

You can run the hdfs related product test in this way:

```
testing/bin/ptl test run --environment multinode  -- -t io.trino.tests.product.hive.TestHdfsSyncPartitionMetadata 
```

However, you can use for the purpose of testing your changes `HiveMinioDataLake` utility.
See for example https://github.com/trinodb/trino/blob/33dd20a8c104d358f5b6d38e8a0405f8b0ced944/plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive3OnDataLake.java#L1764-L1785

Note that you can make use of `io.trino.plugin.hive.containers.HiveMinioDataLake#copyResources` to copy files from the test resources to eventually simulate easier your use case.

Reply:
Thank you for sharing and explaining those..!

I will add some tiny test case on there!

Reply:
I added tiny test case following your guides :)

Reply:
Good job!